beacon_node/beacon_chain/src/custody_context.rs

```diff
@@ -150,6 +150,24 @@ impl ValidatorRegistrations {
                 .or_insert(latest_validator_custody);
         }
     }
+
+    /// Updates the `epoch -> cgc` map by pruning records before `effective_epoch`
+    /// while setting the `cgc` at `effective_epoch` to the latest validator custody requirement.
+    ///
+    /// This is used to restart custody backfill sync at `effective_epoch`
+    pub fn reset_validator_custody_requirements(&mut self, effective_epoch: Epoch) {
```

Comment:
What happens if the latest epoch in `epoch_validator_custody_requirements` is `effective_epoch + 1`?

I _think_ it keeps the `next_epoch` entry in the map, which is probably ok, so we may end up having something like this in the map initially:

```rust
  {
    (current_epoch, 128),
    (next_epoch, 128)
  }
```

and we also overwrite the `DataColumnCustodyInfo` from `next_epoch` to `current_epoch`, and use `current epoch + 1` as backfill start epoch, which also seems fine to me

Reply:
It breaks the assumption here though

https://github.com/sigp/lighthouse/blob/56111a8224381d38d2437806f510ab79f4731f2a/beacon_node/beacon_chain/src/custody_context.rs#L33-L34

Reply:
Addressed in https://github.com/sigp/lighthouse/pull/8339/commits/580b584a8f0838906171a421c3b404ae8a905e07